tests/eclient/testdata/userdata.txt

```diff
@@ -24,9 +26,32 @@ test eden.app.test -test.v -timewait 20m RUNNING eclient
 exec sleep 10
 
 eden pod delete eclient
-eden network delete n1
 
 test eden.app.test -test.v -timewait 10m - eclient
+
+exec -t 10s bash generate_userdata.sh
+eden pod deploy -n eclient --memory=512MB --networks=n1 {{template "eclient_image"}} -p {{$port}}:22 --metadata={{$userdata_file}}
+test eden.app.test -test.v -timewait 20m RUNNING eclient
+
+exec sleep 3
```

Comment:
I would recommend to avoid this sleep and instead make `test_injected_file.sh` to retry few times (with small sleep between retries) until it succeeds or timeouts. Something like [this](https://github.com/lf-edge/eden/blob/master/tests/eclient/testdata/app_local_info.txt#L237-L242) but with smaller sleep time.  Github runners can be slow, meaning that hard-coded sleep time may not be enough (at the same time we do not want to put long unconditional sleeps into tests).

Reply:
thanks! I will do it. it also seems to solve the problem of ssh refusing the connection by simply retrying
```
        > exec -t 2m bash test_injected_file.sh "after_restart"
        [stdout]
        /home/paul/zededa/eden/dist/bin/eden sdn fwd eth0 2223 -- ssh -o ConnectTimeout=10 -o StrictHostKeyChecking=no -i /home/paul/zededa/eden/dist/tests/eclient/image/cert/id_rsa root@FWD_IP -p FWD_PORT grep -q "after_restart" /etc/injected_file.txt
        Try 1
        time="2023-11-06T14:15:39+01:00" level=fatal msg="command ssh failed: exit status 255"
        Try 2
        Success
        [stderr]
        Connection timed out during banner exchange
        Connection to 127.0.0.1 port 2223 timed out
```